pkg/aws/aws_client/client.go

```diff
@@ -35,42 +36,42 @@ type AWSClient struct {
 	KmsClient            *kms.Client
 	CloudWatchLogsClient *cloudwatchlogs.Client
 	AWSConfig            *aws.Config
+	RamClient            *ram.Client
 }
 
 type AccessKeyMod struct {
 	AccessKeyId     string `ini:"aws_access_key_id,omitempty"`
 	SecretAccessKey string `ini:"aws_secret_access_key,omitempty"`
 }
 
-func CreateAWSClient(profileName string, region string) (*AWSClient, error) {
+func CreateAWSClient(profileName string, region string, awsSharedCredentialFile ...string) (*AWSClient, error) {
 	var cfg aws.Config
 	var err error
 
-	if envCredential() {
-		log.LogInfo("Got AWS_ACCESS_KEY_ID env settings, going to build the config with the env")
+	if len(awsSharedCredentialFile) > 0 {
+		file := awsSharedCredentialFile[0]
+		log.LogInfo("Got aws shared credential file path: %s ", file)
 		cfg, err = config.LoadDefaultConfig(context.TODO(),
 			config.WithRegion(region),
-			config.WithCredentialsProvider(
-				credentials.NewStaticCredentialsProvider(
-					os.Getenv("AWS_ACCESS_KEY_ID"),
-					os.Getenv("AWS_SECRET_ACCESS_KEY"),
-					"")),
+			config.WithSharedCredentialsFiles([]string{file}),
 		)
 	} else {
-		if envAwsProfile() {
-			file := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
```

Comment:
You removed the usage of AWS_SHARED_CREDENTIALS_FILE  from the PR. If the ENV is setting AWS_SHARED_CREDENTIALS_FILE then it cannot build up AWS client anymore.